workers/unity/Packages/com.improbable.gdk.core/Config/LocatorConfig.cs

```diff
@@ -55,16 +105,51 @@ public void SetProjectName(string projectName)
         }
 
         /// <summary>
-        ///     Sets the login token for the Locator to use.
+        ///     Sets the login token for the locator to use. Additionally, sets the CredentialsType to LocatorCredentialsType.LoginToken.
         /// </summary>
+        /// <remarks>
+        ///     SpatialOS login token and Steam credentials may not be set at the same time when connecting to a deployment.
+        /// </remarks>
         /// <param name="loginToken">The login token.</param>
+        /// <exception cref="ConnectionFailedException">
+        ///     Thrown if this method is called after Steam credentials were set before.
+        /// </exception>
         public void SetLoginToken(string loginToken)
         {
+            if (LocatorParameters.CredentialsType == LocatorCredentialsType.Steam)
+            {
+                throw new ConnectionFailedException(
+                    "SpatialOS login token and Steam credentials may not be set at the same time when connecting to a deployment.",
+                    ConnectionErrorReason.InvalidConfig);
+            }
+
             LocatorParameters.CredentialsType = LocatorCredentialsType.LoginToken;
-            LocatorParameters.LoginToken = new LoginTokenCredentials
+            LocatorParameters.LoginToken.Token = loginToken;
+        }
+
+        /// <summary>
+        ///     Sets the Steam deployment tag and steam ticket for the locator to use. Additionally, sets the CredentialsType to LocatorCredentialsType.Steam.
+        /// </summary>
+        /// <remarks>
+        ///     SpatialOS login token and Steam credentials may not be set at the same time when connecting to a deployment.
+        /// </remarks>
+        /// <param name="steamDeploymentTag">The deployment tag.</param>
+        /// <param name="steamTicket">The steam ticket.</param>
+        /// <exception cref="ConnectionFailedException">
+        ///     Thrown if this method is called after a SpatialOS login token was set before.
+        /// </exception>
+        public void SetSteamCredentials(string steamDeploymentTag, string steamTicket)
+        {
+            if (LocatorParameters.CredentialsType == LocatorCredentialsType.LoginToken)
             {
-                Token = loginToken
-            };
+                throw new ConnectionFailedException(
+                    "SpatialOS login token and Steam credentials may not be set at the same time when connecting to a deployment.",
```

Comment:
you are using that error message multiple times. factor it out?

Reply:
Done!

Reply:
I'd actually recommend swapping the wording around in each to reflect which one was called first and which one is giving the error. However this is a minor point.

Reply:
I've reworded the error messages to accommodate your suggestion.